app/directives/specification-form/specificationForm.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+(function() {
+    angular.module('proposal').directive('specificationForm',['$location','$state','$http','configProvider','$stateParams','localStorageServiceWrapper',specificationForm]) ;
+    function specificationForm($location,$state,$http,configProvider,$stateParams,localStorageServiceWrapper ) {
+        return {
+          restrict: 'E' ,
+          templateUrl : 'app/directives/specification-form/views/specification-form.html',
+          link:function(scope,element,attr) {          		          		
+          		scope.specification = {} ;
+              //console.log(scope.proposalDetails);
+              if($stateParams.specId && scope.proposalDetails){
+                
+                for(var i=0; i<scope.proposalDetails.specifications.length; i++){
+                  if(scope.proposalDetails.specifications[i].id == $stateParams.specId){
```

Comment:
@pritig-cuelogic @sandeepp-cuelogic @gauravc-cuelogic @tusharncuelogic you can use http://underscorejs.org/ its very easy to use and has many other features.